language-plutus-core/src/Language/PlutusCore/Constant/Make.hs

```diff
@@ -41,6 +48,27 @@ checkBoundsBS size bs = BSL.length bs <= fromIntegral size
 checkBoundsSize :: Size -> Size -> Bool
 checkBoundsSize = (==)
 
+-- | Compute the size of an 'Integer'. See also 'toBoundsInt'.
+sizeOfInteger :: Integer -> Size
+sizeOfInteger i = fromIntegral $ ilogRound 2 (abs i + d) `div` 8 + 1 where
+    d = if i < 0 then 0 else 1
+
+-- | Make a 'Constant' out of an 'Integer'. The size is computed using 'sizeOfInteger'.
+makeAutoSizedBuiltinInt :: Integer -> Constant ()
+makeAutoSizedBuiltinInt i = BuiltinInt () (sizeOfInteger i) i
+
+-- | Convert a Haskell 'Integer' to the corresponding PLC @integer@.
+makeDynamicBuiltinInt
+    :: Type tyname ()       -- ^ An actual size @s@.
+    -> Term tyname name ()  -- ^ A singleton for @s@.
+    -> Integer              -- ^ An 'Integer' to lift.
+    -> Term tyname name ()
+makeDynamicBuiltinInt sTy sTerm intVal =
+    mkIterApp (mkIterInst resizeInteger [TyInt () sizeOfIntVal, sTy]) [sTerm, intTerm] where
+        sizeOfIntVal = sizeOfInteger intVal
```

Comment:
I don't understand why we do this and then resize rather than just specifying it as having the given size to begin with? I guess it makes sense if `resizeInteger` fails...

Reply:
I'm not too happy about it as well. We had a discussion with @psygnisfive about it. Rebecca, could you answer that?

Reply:
> I guess it makes sense if `resizeInteger` fails...

It does fail, but we could have some handy syntax that allows failing as well. Though, the current thing seems to work fine. Especially when I add the `sizeOfInteger` primitive.

Reply:
> rather than just specifying it as having the given size to begin with

Also if the size is a variable, you can't just write `s!0`. A singleton size has to be involved.

Reply:
It would be kind of neat if integer literals evaluated to failure if the integer was outside the stated size. That would also help in dealing with potentially unbounded integers.

Reply:
Except it isn't a variable here, is it (in PLC)? You know it when you're generating the term.

Reply:
> It would be kind of neat if integer literals evaluated to failure if the integer was outside the stated size.

It's how it works right now. Except statically we [throw a parse error](https://github.com/input-output-hk/plutus/blob/6803ba80f7a63d04d9cf4fbdef691cd521c971f8/language-plutus-core/src/Language/PlutusCore/Parser.y#L148) in that case. And dynamically we return `Failure`.

> Except it isn't a variable here, is it (in PLC)? You know it when you're generating the term.

I added some docs: https://github.com/input-output-hk/plutus/pull/199/files#diff-458cc5a5357186c94b34062607c2d212R60

Here is the definition of `succInteger`:

```
/\(s :: size) -> \(ss : size s) (i : integer s) -> addInteger {s} i (resizeInteger {1} {s} ss 1!1)
```

`s` is a type variable.

It's not at all obvious how to define `(+ 1)` is PLC. @psygnisfive, can we add the definition of `(+ 1)` to the spec?